Dockerfile

```diff
@@ -79,6 +79,8 @@ ENV VIRTUALENV_READ_ONLY_APP_DATA=1
 ARG GO=1.15.4
 ARG GO_SHA256=eb61005f0b932c93b424a3a4eaa67d72196c79129d9a3ea8578047683e2c80d5
 ENV PATH=/opt/go/bin:$PATH
+ENV GOCACHE=/tmp
+ENV GOFLAGS=-modcacherw
```

Comment:
hmmm yeah this is probably a good idea -- though `pre-commit` itself handles this, `pre-commit-ci` probably does not.  I should probably fix this from the pre-commit-ci side instead of in this image

could you copy the `rmtree` implementation into this script and see if that works?  you'll probably have to make a custom `TemporaryDirectory` too 🤦.  still think go's decision to make that ro was a really stupid one.  I'll fix it on the pre-commit-ci side: https://github.com/pre-commit/pre-commit/blob/92ce2dcbc375831bab244507e8072a834b761946/pre_commit/util.py#L248-L265

Reply:
Oh wait the rmtree failure I was seeing was coming from pre-commit itself:

```
Traceback (most recent call last):
  File "/venv/lib/python3.8/site-packages/pre_commit/util.py", line 56, in clean_path_on_failure
    yield
  File "/venv/lib/python3.8/site-packages/pre_commit/languages/golang.py", line 88, in install_environment
    rmtree(pkgdir)
  File "/venv/lib/python3.8/site-packages/pre_commit/util.py", line 265, in rmtree
    shutil.rmtree(path, ignore_errors=False, onerror=handle_remove_readonly)
  File "/usr/lib/python3.8/shutil.py", line 715, in rmtree
    _rmtree_safe_fd(fd, path, onerror)
  File "/usr/lib/python3.8/shutil.py", line 652, in _rmtree_safe_fd
    _rmtree_safe_fd(dirfd, fullname, onerror)
  File "/usr/lib/python3.8/shutil.py", line 652, in _rmtree_safe_fd
    _rmtree_safe_fd(dirfd, fullname, onerror)
  File "/usr/lib/python3.8/shutil.py", line 652, in _rmtree_safe_fd
    _rmtree_safe_fd(dirfd, fullname, onerror)
  [Previous line repeated 2 more times]
  File "/usr/lib/python3.8/shutil.py", line 672, in _rmtree_safe_fd
    onerror(os.unlink, fullname, sys.exc_info())
  File "/usr/lib/python3.8/shutil.py", line 670, in _rmtree_safe_fd
    os.unlink(entry.name, dir_fd=topfd)
PermissionError: [Errno 1] Operation not permitted: 'build.yml'
```

Coming from here: https://github.com/pre-commit/pre-commit/blob/92ce2dcbc375831bab244507e8072a834b761946/pre_commit/languages/golang.py#L88 🤔 

So it's failing with `EPERM` instead of `EACCES` :( should we update pre-commit's rmtree?

Reply:
shoot I thought I had fixed this!  yeah that should be updated

Reply:
I was trying to reproduce this to send a PR to pre-commit but I suspect it's related to the fact that /pc is a Docker volume 🤔 

Here's the full path + permission to the `build.yml` file:

```
drwxr-xr-x  1 root root    4096 Dec  5 03:24 /
drwx------  7  501 dialout  224 Dec  5 03:25 /pc
drwx------ 31  501 dialout  992 Dec  5 03:24 /pc/repowbc6thp4
drwxr-xr-x  3  501 dialout   96 Dec  5 03:25 /pc/repowbc6thp4/golangenv-default
drwxr-xr-x  3  501 dialout   96 Dec  5 03:25 /pc/repowbc6thp4/golangenv-default/pkg
drwxr-xr-x  9  501 dialout  288 Dec  5 03:25 /pc/repowbc6thp4/golangenv-default/pkg/mod
drwxr-xr-x  3  501 dialout   96 Dec  5 03:25 /pc/repowbc6thp4/golangenv-default/pkg/mod/4d63.com
dr-x------ 11  501 dialout  352 Dec  5 03:25 /pc/repowbc6thp4/golangenv-default/pkg/mod/4d63.com/gochecknoglobals@v0.0.0-20201008074935-acfc0b28355a
dr-xr-xr-x  3  501 dialout   96 Dec  5 03:25 /pc/repowbc6thp4/golangenv-default/pkg/mod/4d63.com/gochecknoglobals@v0.0.0-20201008074935-acfc0b28355a/.github
dr-xr-xr-x  3  501 dialout   96 Dec  5 03:25 /pc/repowbc6thp4/golangenv-default/pkg/mod/4d63.com/gochecknoglobals@v0.0.0-20201008074935-acfc0b28355a/.github/workflows
-r--r--r--  1  501 dialout  267 Dec  5 03:25 /pc/repowbc6thp4/golangenv-default/pkg/mod/4d63.com/gochecknoglobals@v0.0.0-20201008074935-acfc0b28355a/.github/workflows/build.yml
```

Will work a bit more here. If I can't figure this out I'll open an issue in the pre-commit repo with details.